### DISCUSSION TITLE
Loop audio?

### DISCUSSION THREAD

**Discussion by chook100** [zauberzeug/nicegui#555] [Q&A] [answered]
Hello. This is a fairly simple question, but is there a way to loop audio? (Restarts automatically after the sound ends). I've tried using audio.on('ended') and creating a new ui.audio element, but depending on how long the site is up, can quickly bog down a system since they are not being deleted.

**Comment by falkoschindler** [comment, marked as answer]
That's a good question, @chook100!

The HTML audio tag supports the attribute ["loop"](https://www.w3schools.com/tags/att_audio_loop.asp). So you can simply add a `loop` prop:

```py
ui.audio('https://cdn.pixabay.com/download/audio/2022/02/22/audio_d1718ab41b.mp3').props('loop')
```

But NiceGUI should support this feature natively. I'll add it to the public API.

**Reply by falkoschindler** [reply]
Commit https://github.com/zauberzeug/nicegui/commit/9f8002a92ad11701a63f9c0573d7ea2adadff3a8 introduces a `loop` parameter for `ui.audio` and `ui.video`. It will be part of the upcoming 1.2.0 release.
